[yockcheng] iP

yockcheng opened the pull request:
# LictPro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

LictPro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2425S1/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

### Features:
- [x] Managing tasks
- [x]  Managing deadlines
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

yockcheng committed:
Change Duke.java

yockcheng committed:
Level-1

yockcheng committed:
Level-2

yockcheng committed:
Level-3

yockcheng committed:
Level-4

yockcheng committed:
Added Tests

yockcheng committed:
Add exceptions and made helper methods

yockcheng committed:
Add delete and more exception handling

yockcheng committed:
Add Enums and javadocs

yockcheng committed:
Added saveData method

yockcheng committed:
Merge Level 7 implementation

yockcheng committed:
Configure Deadline to accept dates

yockcheng committed:
Configure Event to accept dates

yockcheng committed:
Merge branch-Level-8 to master

yockcheng committed:
Make code OOP by separating into classes

yockcheng committed:
Organise code into packages

yockcheng committed:
Merge add-gradle-support

yockcheng committed:
Set up Gradle

yockcheng committed:
Add JUnit Tests for Event.java, DateTime.java and Todo.java

yockcheng committed:
Add Javadocs for most of the classes and methods

yockcheng committed:
Edit format to comply with coding standard

yockcheng committed:
Add find command to return matching tasks in list

yockcheng committed:
Merge tag 'A-CodingStandard'

yockcheng committed:
Resolve merge conflicts

yockcheng committed:
Add checkstyle and resolve checkstyle violations

yockcheng committed:
Merge CheckStyle implementation branch

yockcheng committed:
Add GUI, HelloCommand and configured UI to remember last output

yockcheng committed:
Merge branch 'branch-Level-10'

InfinityTwo commented on src/main/java/lict/Lict.java:
```diff
@@ -0,0 +1,72 @@
+    public Lict() {
+        this(DEFAULT_FILE_PATH);
+    }
+    /**
```
This seems to be missing a newline here, would it be better to add it?

InfinityTwo commented on src/main/java/lict/Ui.java:
```diff
@@ -0,0 +1,174 @@
+     * @param filter The keyword to filter the tasks by. If the filter is empty, all tasks will be
+     *     considered as non-matching.
+     */
+    public void showFilteredTasks(TaskList tasks, String filter) {
```
Would it be better to add some empty lines in between portions of code in this method for better readiblity?

InfinityTwo commented on src/main/java/lict/command/EventCommand.java:
```diff
@@ -0,0 +1,76 @@
+     * @throws LictException If the input format is invalid or
+     *     if any required information (description, start, end) is missing.
+     */
+    public String[] extractInfo() throws LictException {
```
Ditto on newlines

InfinityTwo reviewed (commented):
Coding standards generally look good, don't really see any issues.

sam-theman88 commented on src/main/java/lict/Storage.java:
```diff
@@ -0,0 +1,81 @@
+     * @return An {@code ArrayList} of tasks loaded from the file.
+     * @throws LictException If an I/O error occurs while loading tasks.
+     */
+    public ArrayList<Task> load() throws LictException {
```
You can consider using a more specific method name e.g loadTasks

sam-theman88 commented on src/main/java/lict/Ui.java:
Nice job on the use of certain prefixes in the method names to clearly explain the method's function and use cases. "show" for methods with void return types and will print something out to the user. "has" for methods that will be called after a Command is executed and will print respective reaction message to the user.

sam-theman88 commented on src/main/java/lict/Parser.java:
```diff
@@ -0,0 +1,118 @@
+
+            case MARK:
+                c = new MarkCommand(info);
+                break;
+
+            case UNMARK:
+                c = new UnmarkCommand(info);
+                break;
+
+            case DELETE:
+                c = new DeleteCommand(info);
+                break;
+
+            case TODO:
+                c = new TodoCommand(info);
+                break;
+
+            case DEADLINE:
+                c = new DeadlineCommand(info);
+                break;
+
+            case EVENT:
+                c = new EventCommand(info);
+                break;
+
+            case FIND:
+                c = new FindCommand(info);
+                break;
+
+            case HELLO:
+                c = new HelloCommand();
+                break;
+
+            case BYE:
+                c = new ByeCommand();
+                break;
+
+            default:
+                //Fallthrough
+            }
```
Nice use of switch statements and enums here! Great job in following the coding standard for the module too with regards to indentation for the different cases!

sam-theman88 commented on src/test/java/lict/task/EventTest.java:
Might want to consider using the naming convention for writing test methods to clearly show the method being called, the type of input to be provided, and the expected outcome.

sam-theman88 reviewed (commented):
Overall, LGTM! Great job in having clear and concise code quality in terms of following the coding standards and conventions :) Just have some nitpicks. Keep up the good work!

yockcheng committed:
Fix checkstyle issues with EventCommand, Storage and MainWindow

Import statements were not following the correct order.
Some lines of code exceeded the maximum allowed length, making the code harder to read.
Wildcard imports were used, which can lead to unnecessary imports and hinder code readability.

Let's,

*reorder the import statements to follow the correct convention
*break long lines into shorter ones to comply with the line length limit
*replace wildcard imports with specific class imports to improve clarity and maintainability

yockcheng committed:
Add assert statements to validate inputs before using them as arguments

Missing checks such as empty filepath which may cause issues in running the application

Use assert statements as a safety net in case of accidental deletions or mistakes when coding which could lead to errors

Let's add assert statements to ensure that all the essential components are present such as a present filepath to load and save data from, and a valid description for tasks.

yockcheng committed:
Refactor code to improve code quality

yockcheng committed:
Merge branch-A-Assertions with branch-A-CodeQuality

yockcheng committed:
Add basic features for snoozing deadline and event tasks

yockcheng committed:
Fix snooze deadline and event bugs such as whitespace inputs

yockcheng committed:
Add javadocs for Deadline and Event classes

yockcheng committed:
Edit user interface pictures for lict and user

yockcheng committed:
Reconfigure user interface background and text colours to space-like theme

yockcheng committed:
Update snoozeTask in Deadline to allow for accidental whitespaces in user input

yockcheng committed:
Add binary files into .gitignore and remove binary files from main/java/lict

yockcheng committed:
Clean up coding quality using more abstraction

yockcheng committed:
Update Lict user guide with feature details

yockcheng committed:
Fix bugs in test cases for DateTime and Event

yockcheng committed:
Update Lict.jar file link in user guide

yockcheng committed:
Edit expected output in README for Deadline and Event commands
